subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv.c

```diff
@@ -99,6 +99,13 @@ int lll_adv_reset(void)
 	return 0;
 }
 
+#if defined(CONFIG_BT_TX_PWR_DYNAMIC_CTRL)
+void lll_adv_tx_pwr_reset(struct lll_adv *lll)
```

Comment:
You dont need these if you are only reseting the value on state/role teardown. Remove every where, only assign when state/role is instantiated.

Reply:
Implemented TXP defaults at instantiation/setup procedures level.

Advertiser state: https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/a0f3d32567b51755d3f2e8869dfe1e4c3f487a2d#diff-7dd8f83650705279a5ab78ea1e56798dR450

Advertiser state - connection provision: https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/a0f3d32567b51755d3f2e8869dfe1e4c3f487a2d#diff-7dd8f83650705279a5ab78ea1e56798dR604

Scanner state: https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/a0f3d32567b51755d3f2e8869dfe1e4c3f487a2d#diff-5fde7b773bf177bd42dddfd1715205a1R107

Master role: https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/a0f3d32567b51755d3f2e8869dfe1e4c3f487a2d#diff-50a16934d1f6f0c7540768f7c36cffefR147

Slave role: https://github.com/zephyrproject-rtos/zephyr/pull/19779/commits/a0f3d32567b51755d3f2e8869dfe1e4c3f487a2d#diff-1cfc208e525b4f77177d8a72616b84a7R172